### DISCUSSION TITLE
Http Kernel Docs

### DISCUSSION THREAD

**Discussion by NikiforovAll** [dotnet/interactive#3742] [Q&A]
I think HTTP Kernel has a lot of potential, I would like to see more docs for it. Do you accept PRs?

**Comment by jonsequitur** [comment]
We do!

[These docs here](https://learn.microsoft.com/en-us/aspnet/core/test/http-files) describe the language-level features (since Visual Studio uses the HTTP kernel to power the HTTP Editor). The UI and interaction model are a bit different obviously, but a pointer to that page is a good start for people wanting to learn more about this.

**Reply by NikiforovAll** [reply]
It is a good start. But it is missing so many things that makes `.http` files usable. For example, I would like to know how to get a response body from a named request and share it with another kernel.

**Reply by jonsequitur** [reply]
Here's a start. `--byref` seems to be silently failing though. I'll look into that.

![image](https://github.com/user-attachments/assets/739e8d22-249d-42b5-bd2f-bc5b1a85e0f3)

**Reply by NikiforovAll** [reply]
I would like to attach a scenario that happens to me frequently when I use the HTTP kernel. Right now, I'm not sure how to achieve this easily: 

1. Make an HTTP request
2. Get a body of response / or subset of the body using JsonPath
3. Output the value (ideally in the same sell)

I want to do this because the current output requires constant use of the mouse to drill down to each element. I'd like to specify a JSONPath and see the formatted JSON directly in the output.

**Reply by jonsequitur** [reply]
We've been discussing a more compact and usable output. Do you find the Visual Studio HTTP editor more helpful?

**Reply by NikiforovAll** [reply]
I use an HTTP client in vscode, it opens the output to the side as plain/text. So no extra clicks most of the time.

**Reply by jonsequitur** [reply]
I wonder if collapsing the body and expanding the headers by default would be better, as a simple start. This takes up much less space.

![image](https://github.com/user-attachments/assets/3426c5cb-9f97-4b01-a470-5cc50df35224)
